src/Nox.Lib/Application/Queries/IValidateEntityChainQuery.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Nox.Application.Queries;
+
+public interface IValidateEntityChainQueryHandler
```

Comment:
I think handler I think mislead people...

This is actually Relationship Validator, maybe IRelationshipChainValidator?